crates/noirc_frontend/src/parser/parser.rs

```diff
@@ -524,13 +524,12 @@ fn trait_implementation_body() -> impl NoirParser<Vec<TraitImplItem>> {
 }
 
 fn where_clause() -> impl NoirParser<Vec<TraitConstraint>> {
-    let constraints = parse_type()
+    let constraints = ident()
```

Comment:
Lets change this back to `parse_type`, this position of a where clause can be any arbitrary type including `where Vec<T>: Foo`, `where (T, T): Bar`, or `where u32: Bar`.

Reply:
Indeed, Rust supports this - just checked. I didn't know (first month of writing Rust :) ). I will fix this.

Reply:
Fixed.